drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -0,0 +1,583 @@
+/*
+ *
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <soc.h>
+#include <stm32_ll_bus.h>
+#include <stm32_ll_pwr.h>
+#include <stm32_ll_rcc.h>
+#include <stm32_ll_utils.h>
+#include <stm32_ll_system.h>
+#include <drivers/clock_control.h>
+#include <sys/util.h>
+#include <stm32_ll_utils.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+
+/* Macros to fill up prescaler values */
+#define z_ahb_prescaler(v) LL_RCC_SYSCLK_DIV_ ## v
+#define ahb_prescaler(v) z_ahb_prescaler(v)
+
+#define z_apb1_prescaler(v) LL_RCC_APB1_DIV_ ## v
+#define apb1_prescaler(v) z_apb1_prescaler(v)
+
+#define z_apb2_prescaler(v) LL_RCC_APB2_DIV_ ## v
+#define apb2_prescaler(v) z_apb2_prescaler(v)
+
+#define z_apb3_prescaler(v) LL_RCC_APB3_DIV_ ## v
+#define apb3_prescaler(v) z_apb3_prescaler(v)
+
+#if STM32_AHB_PRESCALER > 1
+/*
+ * AHB prescaler allows to set a HCLK frequency (feeding cortex systick)
```

Comment:
HCLK is the core clock it is also SystemCoreClock. So this restriction doesn't apply in this case.

Reply:
Cf https://github.com/zephyrproject-rtos/zephyr/commit/77db273f6f84b3d94d34b5e72070ec6bc31e41cf
I don't see the difference in U5 vs other series

Reply:
As discussed, restriction doesn't apply and should be fixed on at least some other series as well;
Will be fixed globally once this PR is merged.